store/sqlstore/reaction_store.go

```diff
@@ -228,6 +228,132 @@ func (s *SqlReactionStore) PermanentDeleteBatch(endTime int64, limit int64) (int
+	query := `
+		SELECT
+			EmojiName,
+			sum(EmojiCount) AS Count
+		FROM ((
+				SELECT
+					EmojiName,
+					count(EmojiName) AS EmojiCount
+				FROM
+					ChannelMembers
+					INNER JOIN Channels ON ChannelMembers.ChannelId = Channels.Id
+					INNER JOIN Posts ON Channels.Id = Posts.ChannelId
+					INNER JOIN Reactions ON Posts.Id = Reactions.PostId
+				WHERE
+					Reactions.DeleteAt = 0
+					AND ChannelMembers.UserId = ?
+					AND Channels.Type = 'P'
+					AND Channels.TeamId = ?
+					AND Reactions.CreateAt > ?
+				GROUP BY
+					Reactions.EmojiName)
+			UNION ALL (
+				SELECT
+					EmojiName,
+					count(EmojiName) AS EmojiCount
+				FROM
+					Reactions
+					INNER JOIN Posts ON Reactions.PostId = Posts.Id
+					INNER JOIN Channels ON Posts.ChannelId = Channels.Id
+				WHERE
+					Reactions.DeleteAt = 0
+					AND Channels.Type = 'O'
+					AND Channels.TeamId = ?
+					AND Reactions.CreateAt > ?
+				GROUP BY
+					Reactions.EmojiName)) AS A
+		GROUP BY
+			EmojiName
+		ORDER BY
+			Count DESC
+		LIMIT ?
+		OFFSET ?`
+
+	if err := s.GetReplicaX().Select(&reactions, query, userID, teamID, since, teamID, since, limit, offset); err != nil {
+		return nil, errors.Wrap(err, "failed to get top Reactions")
+	}
+	return reactions, nil
+}
+
+func (s *SqlReactionStore) GetTopForUserSince(userID string, teamID string, since int64, offset int, limit int) ([]*model.TopReactions, error) {
+	var reactions []*model.TopReactions
+	var args []interface{}
+	var query string
+
+	if teamID != "" {
+		query = `
+			SELECT
+				EmojiName,
+				sum(EmojiCount) AS Count
+			FROM ((
+					SELECT
+						EmojiName,
+						count(EmojiName) AS EmojiCount
+					FROM
+						Reactions
+						INNER JOIN Posts ON Reactions.PostId = Posts.Id
+						INNER JOIN Channels ON Posts.ChannelId = Channels.Id
+					WHERE
+						Reactions.DeleteAt = 0
+						AND Reactions.UserId = ?
+						AND Channels.TeamId = ?
+						AND Reactions.CreateAt > ?
+					GROUP BY
+						Reactions.EmojiName)
+				UNION ALL (
+					SELECT
+						EmojiName,
+						count(EmojiName) AS EmojiCount
+					FROM
+						Reactions
+						INNER JOIN Posts ON Reactions.PostId = Posts.Id
+						INNER JOIN Channels ON Posts.ChannelId = Channels.Id
+					WHERE
+						Reactions.DeleteAt = 0
+						AND Reactions.UserId = ?
+						AND (Channels.Type = 'D' OR Channels.Type = 'G')
+						AND Reactions.CreateAt > ?
+					GROUP BY
+						Reactions.EmojiName)) AS A
+			GROUP BY
+				EmojiName
+			ORDER BY
+				Count DESC
+			LIMIT ?
+			OFFSET ?`
+		args = []interface{}{userID, teamID, since, userID, since, limit, offset}
+	} else {
+		query = `
+			SELECT
+				EmojiName,
+				count(EmojiName) AS Count
+			FROM
+				Reactions
+			WHERE
+				Reactions.DeleteAt = 0
+				AND Reactions.UserId = ?
+				AND Reactions.CreateAt > ?
+			GROUP BY
+				Reactions.EmojiName
+			ORDER BY
+				Count DESC
+			LIMIT ?
+			OFFSET ?`
+		args = []interface{}{userID, since, limit, offset}
+	}
```

Comment:
Was the `UNION ALL` necessary for this one or could we simplify it to something like:

```sql
SELECT
    EmojiName,
    count(EmojiName) AS Count
        FROM
            Reactions
            INNER JOIN Posts ON Reactions.PostId = Posts.Id
            INNER JOIN Channels ON Posts.ChannelId = Channels.Id
        WHERE
            Reactions.DeleteAt = 0
            AND Reactions.UserId = ?
            AND (Channels.TeamId = ? OR Channels.Type = 'D' OR Channels.Type = 'G')
            AND Reactions.CreateAt > 0
GROUP BY
    EmojiName
ORDER BY
    Count DESC
LIMIT ?
OFFSET ?
```

Reply:
You're right, the `UNION ALL` and additional joins were unnecessary and slowed down the query. Updated to use `OR` instead.